### PR TITLE
swarm: schemas and migrations

### PR DESCRIPTION
**Pull request by nonsense**
The idea of this PR is to add schema identifier to every LevelDB instance, so that we can migrate or cleanup in case Swarm binary runs on a LevelDB instance that it doesn't detect.

Schemas have names, and not versions, so that we think carefully what needs to happen once we have 2-3 different schema in production.

The idea for this first formal schema is to run the `ldbstore.Cleanup()` task on start-up.

This has been tested on the staging cluster and all invalid chunks (according to the Validators we run) are removed.

Addresses https://github.com/ethersphere/go-ethereum/issues/909

### CODE REVIEW

File: swarm/storage/ldbstore.go [hunk 1]
```diff
@@ -730,6 +731,28 @@ func (s *LDBStore) tryAccessIdx(ikey []byte, index *dpaDBIndex) bool {
 	return true
 }
 
+func (s *LDBStore) GetSchema() (string, error) {
```
Comment (zelig) [inline swarm/storage/ldbstore.go]: comment exported functions

File: swarm/storage/ldbstore.go [hunk 2]
```diff
@@ -730,6 +731,28 @@ func (s *LDBStore) tryAccessIdx(ikey []byte, index *dpaDBIndex) bool {
...
+		if err == leveldb.ErrNotFound {
+			return "", nil
+		}
+		return "", err
+	}
+
+	return string(data), nil
+}
+
+func (s *LDBStore) PutSchema(schema string) error {
```
Comment (zelig) [inline swarm/storage/ldbstore.go]: comment exported functions

File: swarm/storage/localstore.go [hunk 3]
```diff
@@ -184,3 +184,40 @@ func (ls *LocalStore) Iterator(from uint64, to uint64, po uint8, f func(Address,
 func (ls *LocalStore) Close() {
 	ls.DbStore.Close()
 }
+
+func (ls *LocalStore) Migrate() error {
```
Comment (zelig) [inline swarm/storage/localstore.go]: comment exported functions

File: swarm/storage/schema.go [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package storage
+
+// "purity" is the first formal schema of LevelDB we release together with Swarm 0.3.5
```
Comment (zelig) [inline swarm/storage/schema.go]: might be 0.4?

Comment (nonsense) [reply swarm/storage/schema.go]: next is 0.3.5 AFAIK, and it looks like we will be merging this before 0.3.5 is released.

**Pull request merged**